{"id": "49503", "title": "Bug 49503: Connectors do not bind to their ports in Catalina.load(), but only later in start()", "description": "Bug 49503: Connectors do not bind to their ports in Catalina.load(), but only later in start(). Testing with trunk at revision 957942, thus this supposedly affects 7.0.0 as well. In TC6 there is a separate prestartup phase when starting Tomcat, to allow Connectors to bind to their ports before other parts of Tomcat are started. IIRC, it is implemented by having separate Bootstrap.load(String[]) and Boot.start() methods [1]. Essentially, it breaks the fix for BZ 48625. To reproduce: 1. The first instance starts successfully.", "OB": "During the startup process of Tomcat, connectors do not bind to their ports in the Catalina.load() method. Instead, they bind to their ports later in the start() method.", "EB": "I would expect that in the StandardService.initInternal() method, the connector.init() call fails because it cannot bind to the port. The exception thrown by connector.init() is caught there and, if the EXITONINITFAILURE flag is set, it is rethrown. Then, it is caught in Catalina.load() and rethrown as an Error, which aborts Tomcat startup.", "SR": "To reproduce the issue, follow these steps: 1. Add the following line to conf/catalina.properties: org.apache.catalina.startup.EXITONINITFAILURE=true. 2. Start two Tomcat instances with the same configuration at the same time."}